arch/src/x86_64/mod.rs

```diff
@@ -200,6 +208,24 @@ fn configure_pvh(
         start_info.0.rsdp_paddr = rsdp_addr.0;
     }
 
+    if let Some(initramfs_config) = initramfs {
+        // The initramfs has been written to guest memory already, here we just need to
+        // create the module structure that describes it.
+        let ramdisk_mod: ModlistEntryWrapper = ModlistEntryWrapper(hvm_modlist_entry {
+            paddr: initramfs_config.address.raw_value(),
+            size: initramfs_config.size as u64,
+            ..Default::default()
+        });
+
+        start_info.0.nr_modules += 1;
+        start_info.0.modlist_paddr = layout::MODLIST_START.raw_value();
+
+        // Write the modlist struct to guest memory.
+        guest_mem
+            .write_obj(ramdisk_mod, layout::MODLIST_START)
+            .map_err(|_| super::Error::ModlistSetup)?;
```

Comment:
Can you make `ModlistSetup` take the error from `.write_obj`

Reply:
Yes, should be fixed now.

Reply:
Added integration test too.